### PR TITLE
Ticket 11 - GET /api/articles (topic query)

### PR DESCRIPTION
**Pull request by VenturingForth**
CORE: GET /api/articles (topic query)

Testing
200: Should retrieve an array of article objects filtered by topic with the necessary properties.
200: Should retrieve an empty array if topic is in database, but does not associated with an article.
404: Should return "Topic not found" if topic is not in the database

Documentation:
Endpoints JSON updated.

### CODE REVIEW

File: models/articles-models.js [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
-        `).
-    then(({rows}) => {
+        `
+
+    if(topic){
+        queryStr += ` WHERE topic = $1`;
+        queryValues.push(topic);
+    }
+
+    queryStr += ` GROUP BY articles.author, articles.title, articles.article_id ORDER BY created_at DESC;`
```
Comment (nc-katherine-hurst) [inline models/articles-models.js:17]: Nice use of string concatenation to adjust the query only when a topic has been provided 🙌

File: controllers/articles-controllers.js [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
-    return fetchArticles().then((articles) => {
-        res.status(200).send({articles});
-    }).catch(next);
+    if (Object.keys(req.query).length !== 0){
+        const { topic } = req.query;
+        const checkTopicExistsQuery = checkTopicExists(topic);
+        const articlesQuery = fetchArticles(topic);
+        const queryArray = [articlesQuery, checkTopicExistsQuery];
+        return Promise.all(queryArray)
+        .then((articles) => {
```
Comment (nc-katherine-hurst) [inline controllers/articles-controllers.js:16]: I think you should rename this `articles` variable as it describes an array where only the first value are the articles 🔍

File: __tests__/app.test.js [hunk 3]
```diff
@@ -71,6 +71,35 @@ describe("GET /api", () => {
...
+            })
+            test("404: Should return 'Topic not found' if topic is not in the database.", () => {
+                return request(app)
+                .get('/api/articles?topic=joshua')
+                .expect(404)
+                .then(({body}) => {
+                    expect(body.msg).toBe("Topic not found");
+                })
+            })
+        })
```
Comment (nc-katherine-hurst) [inline __tests__/app.test.js:102]: Nice error handling here, covered all the bases 🎯

**Review by nc-katherine-hurst — COMMENTED** [review]
Hi Josh, nice work please have a look at my comments below 👏

File: controllers/articles-controllers.js (hunk 2 above)
Comment (VenturingForth) [reply controllers/articles-controllers.js:16]: I've renamed this variable to make it more clear that it is the results of the Promise.all and that the articles list is the first result of that array. Refactored.

**Pull request merged**